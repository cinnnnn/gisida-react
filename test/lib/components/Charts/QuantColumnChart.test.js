import React from 'react';
import { QuantColumnChart }  from '../../../../src/lib/components/Charts/QuantColumnChart.jsx'
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

const componentWrapper = mount(
	<QuantColumnChart 
		data={[]}
		passing={[]}
	/>
);

const data = [10,20,30]
const passing = ['Jan','Feb','Mar']

describe('QuantColumnChart', () => {

	it('QuantColumnChart component renders correctly', () => {
		const json = toJson(componentWrapper)
		expect(json).toMatchSnapshot();
	})

	it('should set state', () => {
		componentWrapper.setState({ show: true });
		expect(componentWrapper.state('show')).toEqual(true);
	})

	it('should set props', () => {
		const propsUpdateMock = jest.spyOn(componentWrapper.instance(), 'componentWillReceiveProps');
		componentWrapper.setProps({data, passing});
		expect(componentWrapper.props().data).toEqual(data);
		expect(propsUpdateMock).toHaveBeenCalledTimes(1);
	})
});