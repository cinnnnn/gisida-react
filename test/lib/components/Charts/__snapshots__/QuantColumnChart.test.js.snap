// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantColumnChart QuantColumnChart component renders correctly 1`] = `
<QuantColumnChart
  data={Array []}
  passing={Array []}
>
  <ColumnChart
    categories={Array []}
    chartHeight={150}
    chartWidth={230}
    isPercent={false}
    pointFormatterFunc={[Function]}
    seriesData={Array []}
    yAxis={
      Object {
        "endOnTick": false,
        "visible": false,
      }
    }
  >
    <div />
  </ColumnChart>
</QuantColumnChart>
`;
